tools/plugins/com.liferay.ide.ui/src/com/liferay/ide/ui/LiferayUIPlugin.java

```diff
@@ -270,6 +277,16 @@ protected void registerImage(ImageRegistry registry, String key, String path) {
 	protected TextFileDocumentProvider fTextFileDocumentProvider;
 	protected Map<String, ImageDescriptor> imageDescriptors = new HashMap<>();
 
+	private void _addRepository() throws URISyntaxException {
+		ProvisioningUI provisioningUI = ProvisioningUI.getDefaultUI();
+
+		ProvisioningSession provisioningSession = provisioningUI.getSession();
+
+		URI uri = new URI("https://releases.liferay.com/tools/ide/latest/stable/");
+
+		provisioningUI.getRepositoryTracker().addRepository(uri, "Liferay IDE Stable releases", provisioningSession);
```

Comment:
hey @JoyeLuo  can you send a new pull request with the following?

1) Do we need to check if the user already contains a repository with the same URI?  What if the repository tracker already has a repo?
2) Can we remove the chaining here?  `provisioningUI.getRepositoryTracker()` `repositoryTracker.addRepository()`

Reply:
Thanks, got it. Will send a new pr.

Reply:
@JoyeLuo can you also add the base ui test for it? I opened the ticket:https://issues.liferay.com/browse/IDE-3610 , and please let @yanan know how it work, thanks.

Reply:
Ok, got it.